src/orion/algo/space/configspace.py

```diff
@@ -0,0 +1,262 @@
+        if dim.prior_name in ("reciprocal", "uniform"):
+            a, b = dim._args
+
+            return UniformFloatHyperparameter(
+                name=dim.name,
+                lower=a,
+                upper=b,
+                default_value=dim.default_value,
+                q=_qantization(dim),
+                log=dim.prior_name == "reciprocal",
+            )
+
+        if dim.prior_name in ("normal", "norm"):
+            a, b = dim._args
+
+            kwargs = dict(
+                name=dim.name,
+                mu=a,
+                sigma=b,
+                default_value=dim.default_value,
+                q=_qantization(dim),
+                log=False,
+                lower=dim.low if hasattr(dim, "low") else None,
+                upper=dim.high if hasattr(dim, "high") else None,
+            )
+
+            return NormalFloatHyperparameter(**kwargs)
+
+        raise UnsupportedPrior(f'Prior "{dim.prior_name}" is not supported')
+
+    def integer(self, dim: Integer) -> IntegerHyperparameter:
+        """Convert a integer dimension into a configspace equivalent"""
+        if dim.prior_name in ("int_uniform", "int_reciprocal"):
+            a, b = dim._args
+
+            return UniformIntegerHyperparameter(
+                name=dim.name,
+                lower=a,
+                upper=b,
+                default_value=dim.default_value,
+                q=_qantization(dim),
+                log=dim.prior_name == "int_reciprocal",
+            )
+
+        if dim.prior_name in ("int_norm", "normal"):
+            a, b = dim._args
+
+            kwargs = dict(
+                name=dim.name,
+                mu=a,
+                sigma=b,
+                default_value=dim.default_value,
+                q=_qantization(dim),
+                log=False,
+                lower=dim.low if hasattr(dim, "low") else None,
+                upper=dim.high if hasattr(dim, "high") else None,
+            )
+
+            return NormalIntegerHyperparameter(**kwargs)
+
+        raise UnsupportedPrior(f'Prior "{dim.prior_name}" is not supported')
+
+    def categorical(self, dim: Categorical) -> CategoricalHyperparameter:
+        """Convert a categorical dimension into a configspace equivalent"""
+        return CategoricalHyperparameter(
+            name=dim.name,
+            choices=dim.categories,
+            weights=dim._probs,
+        )
+
+    def fidelity(self, dim: Fidelity) -> None:
+        """Ignores fidelity dimension as configspace does not have an equivalent"""
+        return None
+
+    def space(self, space: Space) -> ConfigurationSpace:
+        """Convert orion space to configspace"""
+        cspace = ConfigurationSpace()
+        dims = []
+
+        for _, dim in space.items():
+            cdim = self.convert_dimension(dim)
+
+            if cdim:
+                dims.append(cdim)
+
+        cspace.add_hyperparameters(dims)
+        return cspace
+
+
```

Comment:
Make this a singedispatch callable, and copy-paste the methods of the `ToConfigSpace` class into handler functions for each supported dimension type below. 
```suggestion
@singledispatch
```